### DISCUSSION TITLE
Permit2_permit

### DISCUSSION THREAD

**Discussion by gauravsc** [Elnaril/uniswap-universal-router-decoder#16] [Q&A] [answered]
Thank you for this very useful library. I am wondering if we should put the real uniswap universal router address in the `spender` variable inside the call to permit2_permit (below) or the `FunctionRecipient.ROUTER`?  

```
data, signable_message = codec.create_permit2_signable_message(
    token_address,
    amount,  # max = 2**160 - 1
    expiration,
    nonce,  # Permit2 nonce
    spender,  # UR
    deadline,
    1,  # chain id
)
```

**Comment by Elnaril** [comment, marked as answer]
Hello @gauravsc 
It's the actual UR checksum address, but you're right, it can be confusing. I'm going to clarify it in the README.
Thank you for your feedback! :)
Happy coding!

**Comment by lapkarich** [comment]
Thank you for this very useful library.

**Reply by Elnaril** [reply]
Thank you for your feedback, it is really appreciated! :)

**Comment by marik567** [comment]
cool

**Comment by BritvaBo** [comment]
thanks for review

**Comment by svet345** [comment]
super

**Comment by pufik111** [comment]
good
